into/backends/sparksql.py

```diff
@@ -31,6 +33,19 @@ def iterable_to_sql_context(ctx, seq, **kwargs):
     return append(ctx, append(ctx._sc, seq, **kwargs), **kwargs)
 
 
+@append.register(SQLContext, (JSONLines, Directory(JSONLines)))
+def jsonlines_to_sparksql(ctx, json, dshape=None, name=None, schema=None,
+                          samplingRatio=0.25, **kwargs):
+    # if we're passing in schema, assume that we know what we're doing and
+    # bypass any automated dshape inference
+    if dshape is not None and schema is None:
+        schema = dshape_to_schema(dshape.measure
+                                  if isrecord(dshape.measure) else dshape)
```

Comment:
I didn't mean to scare away schema.  My last comment was "oh, unless this
is a sparksql thing"

I thought it might be a holdover from my previous code when we used to use
schema as the dshape of a record.  If a backend library uses it then it's
fine with me.

On Fri, Feb 13, 2015 at 4:48 PM, Phillip Cloud <notifications@github.com>
wrote:

> In into/backends/sparksql.py
> <https://github.com/ContinuumIO/into/pull/97#discussion_r24709139>:
>
> > @@ -31,6 +33,19 @@ def iterable_to_sql_context(ctx, seq, **kwargs):
> >      return append(ctx, append(ctx._sc, seq, **kwargs), **kwargs)
> >
> >
> > +@append.register(SQLContext, (JSONLines, Directory(JSONLines)))
> > +def jsonlines_to_sparksql(ctx, json, dshape=None, name=None, schema=None,
> > +                          samplingRatio=0.25, **kwargs):
> > +    # if we're passing in schema, assume that we know what we're doing and
> > +    # bypass any automated dshape inference
> > +    if dshape is not None and schema is None:
> > +        schema = dshape_to_schema(dshape.measure
> > +                                  if isrecord(dshape.measure) else dshape)
>
> i'm ok with leaving schema out for now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/into/pull/97/files#r24709139>.
>